packages/twenty-front/src/modules/workflow/workflow-steps/workflow-actions/email-action/extensions/resizable-image/ResizableImageView.tsx

```diff
@@ -0,0 +1,207 @@
+  top: 50%;
+  transform: translateY(-50%);
+  width: ${({ theme }) => theme.spacing(2)};
+  z-index: 1;
+
+  ${({ handle }) => {
+    if (handle === 'left') {
+      return css`
+        left: 2px;
+      `;
+    }
+
+    return css`
+      right: 2px;
+    `;
+  }}
+`;
+
+type ResizeParams = {
+  initialWidth: number;
+  initialClientX: number;
+  handleUsed: 'left' | 'right';
+};
+
+type ResizableImageViewProps = NodeViewProps;
+
+export const ResizableImageView = (props: ResizableImageViewProps) => {
+  const { editor, node, updateAttributes } = props;
+  const { width: initialWidth, align = 'left', src, alt } = node.attrs;
+
+  const imageWrapperRef = useRef<HTMLDivElement>(null);
+
+  const [isDragging, setIsDragging] = useState(false);
+  const [width, setWidth] = useState(initialWidth || 0);
+  const [resizeParams, setResizeParams] = useState<ResizeParams | null>(null);
+
+  const handleMouseMove = useCallback(
+    (event: MouseEvent) => {
+      const imageWrapper = imageWrapperRef.current;
+      if (!isDefined(resizeParams) || !isDefined(imageWrapper)) {
+        return;
+      }
+
+      const deltaX = event.clientX - resizeParams.initialClientX;
+      const { initialWidth, handleUsed } = resizeParams;
+
+      let newWidth =
+        align === 'center'
+          ? initialWidth + (handleUsed === 'left' ? -deltaX * 2 : deltaX * 2)
+          : initialWidth + (handleUsed === 'left' ? -deltaX : deltaX);
+
+      const maxWidth =
+        editor.view.dom.firstElementChild?.clientWidth || IMAGE_MAX_WIDTH;
+      newWidth = Math.min(Math.max(newWidth, IMAGE_MIN_WIDTH), maxWidth);
+
+      setWidth(newWidth);
+      imageWrapper.style.width = `${newWidth}px`;
+    },
+    [resizeParams, editor, align],
+  );
+
+  const handleMouseUp = useCallback(
+    (event: MouseEvent) => {
+      const imageWrapper = imageWrapperRef.current;
+      if (!isDefined(imageWrapper) || !isDefined(resizeParams)) {
+        return;
+      }
+
+      if (
+        (!event.target || !imageWrapper.contains(event.target as Node)) &&
+        isDragging
+      ) {
+        setIsDragging(false);
+        return;
+      }
+
+      setResizeParams(null);
+      updateAttributes({ width });
+    },
+    [resizeParams, updateAttributes, width, isDragging, setIsDragging],
+  );
+
+  const handleImageHandleMouseDown = useCallback(
+    (handle: 'left' | 'right', event: React.MouseEvent<HTMLDivElement>) => {
+      event.preventDefault();
+
+      setResizeParams({
+        initialWidth: imageWrapperRef.current?.clientWidth ?? IMAGE_MAX_WIDTH,
+        initialClientX: event.clientX,
+        handleUsed: handle,
+      });
+    },
+    [],
+  );
+
+  const handleWrapperMouseEnter = useCallback(() => {
+    if (!editor.isEditable) {
+      return;
+    }
+
+    setIsDragging(true);
+  }, [setIsDragging, editor.isEditable]);
+
+  const handleWrapperMouseLeave = useCallback(() => {
+    setIsDragging(false);
+  }, [setIsDragging]);
+
+  useEffect(() => {
```

Comment:
This violates the best practise of avoiding useEffect whenever there is an alternative (there is here imo)

Reply:
Also here you probably recreate event listeners on every render because handleMouseUp/down will change du to their depedencies